lib/ecto/adapters/sql.ex

```diff
@@ -572,6 +572,17 @@ defmodule Ecto.Adapters.SQL do
       def explain(operation, queryable, opts \\ []) do
         Ecto.Adapters.SQL.explain(get_dynamic_repo(), operation, queryable, opts)
       end
+
+      @doc """
+      A convenience function for DBConnection based repositories that forces all connections in the
+      pool to disconnect within the given interval.
+
+      See `DBConnection.disconnect_all/3` for more information.
+      """
```

Comment:
We should also add this to `Ecto.Adapters.SQL.disconnect_all/3`, mostly for discoverability purposes. See how it is done for explain! I would also copy the docs from DBConnection, especially that it doesn't disconnect immediately, and it waits until connections are pinged or are checked in.

We also need a small unit test, simply one that calling the function works is enough?

Reply:
I tried to add an unit test, but it seems that DBConnection.disconnect_all/3 crashes the sandbox (should it do a no-op instead? Maybe try to disconnect only the owned connections? I can send a PR, but ain't sure).